core/command_line_parser.cpp

```diff
@@ -0,0 +1,1076 @@
+			return ParsedPrefix{ _long_prefixes[i], false };
+		}
+	}
+	for (int i = 0; i < _short_prefixes.size(); ++i) {
+		if (p_arg.begins_with(_short_prefixes[i])) {
+			return ParsedPrefix{ _short_prefixes[i], true };
+		}
+	}
+
+	return ParsedPrefix();
+}
+
+String CommandLineParser::_find_most_similar(const String &p_name) const {
+	// Do not search for short names.
+	if (p_name.length() == 1) {
+		return String();
+	}
+
+	String most_similar;
+	float max_similarity = _similarity_bias; // Start with this value to avoid returning unrelated names.
+
+	for (int i = 0; i < _options.size(); ++i) {
+		const PackedStringArray flags = _options[i]->get_names();
+		for (int j = 0; j < flags.size(); ++j) {
+			float similarity = flags[j].similarity(p_name);
+			if (max_similarity < similarity) {
+				most_similar = flags[j];
+				max_similarity = similarity;
+			}
+		}
+	}
+
+	return most_similar;
+}
+
+bool CommandLineParser::_contains_optional_options(const Vector<Pair<const CommandLineOption *, String>> &p_printable_options) {
+	for (int i = 0; i < p_printable_options.size(); ++i) {
+		if (!p_printable_options[i].first->is_required()) {
+			return true;
+		}
+	}
+	return false;
+}
+
+void CommandLineParser::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("parse_args", "args"), &CommandLineParser::parse_args);
+	ClassDB::bind_method(D_METHOD("validate"), &CommandLineParser::validate);
+	ClassDB::bind_method(D_METHOD("add_option", "option"), &CommandLineParser::add_option);
+	ClassDB::bind_method(D_METHOD("get_option_count"), &CommandLineParser::get_option_count);
+	ClassDB::bind_method(D_METHOD("get_option", "idx"), &CommandLineParser::get_option);
+	ClassDB::bind_method(D_METHOD("set_option", "idx", "option"), &CommandLineParser::set_option);
+	ClassDB::bind_method(D_METHOD("remove_option", "idx"), &CommandLineParser::remove_option);
+	ClassDB::bind_method(D_METHOD("find_option", "name"), &CommandLineParser::find_option);
+	ClassDB::bind_method(D_METHOD("add_help_option"), &CommandLineParser::add_help_option);
+	ClassDB::bind_method(D_METHOD("add_version_option"), &CommandLineParser::add_version_option);
+	ClassDB::bind_method(D_METHOD("is_set", "option"), &CommandLineParser::is_set);
+	ClassDB::bind_method(D_METHOD("get_value", "option"), &CommandLineParser::get_value);
+	ClassDB::bind_method(D_METHOD("get_values", "option"), &CommandLineParser::get_values);
+	ClassDB::bind_method(D_METHOD("get_prefix", "option"), &CommandLineParser::get_prefix);
+	ClassDB::bind_method(D_METHOD("get_prefixes", "option"), &CommandLineParser::get_prefixes);
+	ClassDB::bind_method(D_METHOD("get_occurence_count", "option"), &CommandLineParser::get_occurence_count);
+	ClassDB::bind_method(D_METHOD("get_forwarded_args"), &CommandLineParser::get_forwarded_args);
+	ClassDB::bind_method(D_METHOD("get_args"), &CommandLineParser::get_args);
+	ClassDB::bind_method(D_METHOD("get_help_text", "format"), &CommandLineParser::get_help_text, DEFVAL(memnew(CommandLineHelpFormat)));
```

Comment:
This is how it's usually done in Godot:

```suggestion
	ClassDB::bind_method(D_METHOD("get_help_text", "format"), &CommandLineParser::get_help_text, DEFVAL(Variant()));
```

See rationale in my review for `get_help_text()` below.

Reply:
> This is how it's usually done in Godot:

Will it work for `Ref<CommandLineHelpFormat>`?

Reply:
Oh, got it, probably will better to switch to `Variant`.